### PR TITLE
chore(angular): upgrade to official 18 release

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

Angular and angular standalone starters are using v17.

## What is the new behavior?

- Updated to the official release 18.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

N/A

### CODE REVIEW

File: angular-standalone/base/package.json [hunk 1]
```diff
@@ -52,6 +52,6 @@
     "karma-coverage": "~2.2.0",
     "karma-jasmine": "~5.1.0",
     "karma-jasmine-html-reporter": "~2.1.0",
-    "typescript": "~5.2.2"
+    "typescript": "~5.4.0"
```
Comment (sean-perkins) [inline angular-standalone/base/package.json:55]: Our test app is using `^5.4.0` vs. `~5.4.0`. Do you have a recommendation on why one is more appropriate for the test app vs. the starters?

This applies to both `package.json` diffs.

Comment (thetaPC) [reply angular-standalone/base/package.json:55]: No recommendation. I was simply following the structure of the starters. I also don't have any strong feelings. What is your suggestion here? `^5.4.0` or `~5.4.0`?

Comment (brandyscarney) [reply angular-standalone/base/package.json:55]: The answer to this depends on whether or not we want people to always get the latest patch release or the latest minor release for `typescript`. It looks like typescript [doesn't follow SemVer](https://github.com/microsoft/TypeScript/issues/56916) and pushes breaking changes in minor versions so we are probably safer sticking to patches.

**Review by brandyscarney — APPROVED** [review]

File: angular-standalone/base/package.json (hunk 1 above)
Comment (thetaPC) [reply angular-standalone/base/package.json:55]: @sean-perkins  Going to merge but we can always adjust this in another PR.

**Pull request merged**